### PR TITLE
chore(j-s): Use Table component for past cases

### PR DESCRIPTION
**Pull request by oddsson**
# Use Table component for past cases

[Asana](https://app.asana.com/0/1199153462262248/1206918814427645/f)

## What

Refactor the past cases table to use the new Table component. 

## Why

All tabels should use this component.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Simplified `PastCasesTable` component interface by removing the loading prop.
	- Enhanced case handling and rendering in the `Table` component with improved click handling and modularized rendering logic.

- **Bug Fixes**
	- Improved readability and maintainability of context menu items and case rendering logic.

- **Tests**
	- Streamlined test suite for the `Cases` component by removing unnecessary sorting tests while retaining core functionality tests.
	- Enhanced `Table` component tests with new sorting functionality and improved structure.

- **Chores**
	- Removed deprecated `MobilePastCase` component to streamline the codebase.

### CODE REVIEW

File: apps/judicial-system/web/src/components/Table/PastCasesTable/PastCasesTable.tsx [hunk 1]
```diff
@@ -96,176 +57,112 @@ const PastCasesTable: FC<Props> = ({ cases, loading = false, testid }) => {
...
+                  <Box
+                    marginRight={row.appealState ? 1 : 0}
+                    marginBottom={row.appealState ? 1 : 0}
+                  >
+                    {isIndictmentCase(row.type) ? (
+                      <CaseTag
+                        color={indictmentCaseStateTag.color}
+                        text={formatMessage(indictmentCaseStateTag.text)}
+                      />
+                    ) : (
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/Table/PastCasesTable/PastCasesTable.tsx:131]: They do indeed exist in the CaseListEntry interface

File: apps/judicial-system/web/src/components/Table/PastCasesTable/PastCasesTable.tsx [hunk 2]
```diff
@@ -96,176 +57,112 @@
...
+                      <TagCaseState
+                        caseState={row.state}
+                        caseType={row.type}
+                        isCourtRole={isDistrictCourtUser(user)}
+                        isValidToDateInThePast={row.isValidToDateInThePast}
+                        indictmentRulingDecision={row.indictmentRulingDecision}
+                        indictmentDecision={row.indictmentDecision}
+                      />
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/Table/PastCasesTable/PastCasesTable.tsx:139]: Could you explain further?

File: apps/judicial-system/web/src/components/Table/PastCasesTable/PastCasesTable.tsx [hunk 3]
```diff
@@ -96,176 +57,112 @@
...
+          },
+          {
+            cell: (row) => <CreatedDate created={row.created} />,
+          },
+          {
+            cell: (row) => {
+              const indictmentCaseStateTag = getIndictmentCaseStateTag(
+                row as Case,
+                user,
+              )
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/Table/PastCasesTable/PastCasesTable.tsx]: Nice, thanks 🙏

File: apps/judicial-system/web/src/components/Table/PastCasesTable/PastCasesTable.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/Table/PastCasesTable/PastCasesTable.tsx:139]: Yes, I've verified that these properties do in fact exist in the CaseListEntry interface.

File: apps/judicial-system/web/src/components/Table/Table.spec.tsx [hunk 4]
```diff
@@ -77,15 +89,64 @@ describe('Table', () => {
...
+    const tableRows = await screen.findAllByTestId('tableRow')
+    expect(tableRows[0]).toHaveTextContent('Bono Stingsson')
+    expect(tableRows[1]).toHaveTextContent('Jon Harring Sr.')
+
+    await clickButtonByTestId('defendantsSortButton')
+
+    const tableRows2 = await screen.findAllByTestId('tableRow')
+    expect(tableRows2[0]).toHaveTextContent('Jon Harring Sr.')
+    expect(tableRows2[1]).toHaveTextContent('Bono Stingsson')
+  })
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/Table/Table.spec.tsx:151]: No thanks

**Review by gudjong — APPROVED** [review]
LGTM

**Pull request merged**